src/hibernate-resume/hibernate-resume.c

```diff
@@ -94,6 +106,30 @@ static int setup_hibernate_info_and_warn(void) {
         return 1;
 }
 
+static int action_clear_efi(void) {
+        _cleanup_(efi_hibernate_location_freep) EFIHibernateLocation *e = NULL;
+        int r;
+
+        assert(arg_clear_efi);
+
+        if (in_initrd()) {
+                log_info("Running in initrd, --clear-efi has no effect, exiting.");
+                return 0;
+        }
+
+        /* Let's insist that the system identifier is verified still. After all if things don't match,
+         * the resume wouldn't get triggered in the first place. We should not erase the var if booted
+         * from LiveCD/portable systems/... */
```

Comment:
After re-reading your reply in the above and this comment, yeah, that sounds reasonable.